libs/portals/admin/ids-admin/src/components/Applications/Applications.tsx

```diff
@@ -1,91 +1,233 @@
 import {
   Box,
   Button,
+  Filter,
+  FilterInput,
   GridContainer,
   GridRow,
   Stack,
   Tag,
   Text,
 } from '@island.is/island-ui/core'
-import { Link, useParams } from 'react-router-dom'
-import MockApplications from '../../lib/MockApplications'
-import * as styles from '../TenantsList/TenantsList.css'
+import { Link, useLoaderData, useParams } from 'react-router-dom'
+import React, { useEffect, useState } from 'react'
 import { useLocale } from '@island.is/localization'
 import { m } from '../../lib/messages'
 import { useTenant } from '../../screens/Tenant'
-import { useEffect } from 'react'
 import { replaceParams } from '@island.is/react-spa/shared'
 import { IDSAdminPaths } from '../../lib/paths'
+import * as styles from './Applications.css'
+import { AuthApplicationsList } from './Applications.loader'
 
 const Applications = () => {
+  const originalApplications = useLoaderData() as AuthApplicationsList
   const { tenant } = useParams()
   const { formatMessage } = useLocale()
   const { setNavTitle } = useTenant()
 
+  const [applications, setApplications] = useState<AuthApplicationsList>(
+    originalApplications,
+  )
+  const [inputSearchValue, setInputSearchValue] = useState<string>('')
+
   useEffect(() => {
-    // TODO: Get application by id from backend
     setNavTitle(tenant ? tenant : formatMessage(m.tenants))
   })
 
-  return (
-    <GridContainer>
-      <GridRow rowGap={3}>
-        <Box display={'flex'} justifyContent={'spaceBetween'} columnGap={4}>
-          <Box>
+  const handleSearch = (value: string) => {
+    setInputSearchValue(value)
+
+    if (value.length > 0) {
+      const filteredList = applications.filter((application) => {
+        return (
+          application?.defaultEnvironment.displayName[0].value
+            .toLowerCase()
+            .includes(value.toLowerCase()) ||
+          application.applicationId.toLowerCase().includes(value.toLowerCase())
+        )
+      })
+      setApplications(filteredList)
+    } else {
+      setApplications(originalApplications)
+    }
+  }
+
+  const getHeader = ({ withCreateButton = true }) => {
```

Comment:
you should make it optional, i.e.
`const getHeader = ({ withCreateButton = true } = {}) => {`

or just better to have it as optional argument instead of an object :)